src/vkscript/command_parser.cc

```diff
@@ -789,20 +789,18 @@ Result CommandParser::ProcessUniform() {
   if (!r.IsSuccess())
     return r;
 
-  // Multiply by the input needed because the value count will use the needed
-  // input as the multiplier
   uint32_t value_count = ((cmd->GetOffset() / buf->GetFormat()->SizeInBytes()) *
                           buf->GetFormat()->InputNeededPerElement()) +
                          static_cast<uint32_t>(values.size());
-  // The buffer should only be resized to become bigger. This means that if a
-  // command was run to set the buffer size we'll honour that size until a
-  // request happens to make the buffer bigger.
-  if (value_count > buf->ValueCount())
-    buf->SetValueCount(value_count);
-
-  // Even if the value count doesn't change, the buffer is still resized
-  // because this maybe the first time data is set into the buffer.
-  buf->SetSizeInBytes(buf->GetSizeInBytes());
+  uint32_t element_count = value_count;
```

Comment:
This is a repeat of the code sequence at the SSBO routine.  That's a code smell that there's a common functionality should be refactored.  By the look of it, it seems like all this should be a method on the buffer itself.
In fact, after some digging, there is a similar calculation in Buffer::SetDataWithOffset.
(Hmmm.  But that seems to not accomodate the Pack case... ?)

Suggest defining
Buffer::ResizeForDataWithOffset(const std::vector<Value>& data, uint32_t offset);
This mirrors Buffer::SetDataWithOffset.

Reply:
It's not really resizing though. It is setting a fake size to be used in Dawn buffer creation.